.github/workflows/build-and-test.yml

```
@@ -0,0 +1,41 @@
+name: Build and Test
+
+on: [push, pull_request]
```

```suggestion
on:
  workflow_dispatch:
  pull_request:
    branches:
      - master
  push:
    branches:
      - master
```

When running on all pushes, we will have two sets of pipelines in each PR. Adding the option to manually run the workflow also seems like a good idea to me.